src/main/java/donkeychat/Ui.java

```diff
@@ -0,0 +1,76 @@
+package donkeychat;
+
+public class Ui {
+
+    public void displayIntro() {
+        System.out.println("Hello! I'm DonkeyChat!\nWhat can I do for you?");
+    }
+
+    public void displayBye() {
+        System.out.println("Adios!");
+    }
+
+    public void displayTasks(TaskList taskList) {
+        if (taskList.getSize() == 0) {
+            System.out.println("You have no tasks in your list!");
+            return;
+        }
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskList.getSize(); i++) {
+            Task task = taskList.getAtIndex(i);
+            System.out.println(i + 1 + "." + task);
+        }
+    }
+
+    public void displayMatchingTasks(TaskList taskList, String toMatch) {
+        if (taskList.getSize() == 0) {
+            System.out.println("You have no tasks in your list!");
+            return;
+        }
+        boolean hasFoundMatch = false;
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < taskList.getSize(); i++) {
+            Task task = taskList.getAtIndex(i);
+            if(task.description.contains(toMatch)) {
+                System.out.println(i + 1 + "." + task);
+                hasFoundMatch = true;
+            }
+        }
+        if(!hasFoundMatch) {
+            System.out.println("No Tasks with matching Description found!");
+        }
+    }
+
+    public void displayText(String text) {
+        System.out.println(text);
+    }
+
+    public void displayMarkTask(boolean marked, Task task) {
+        if (true) {
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(task);
+        } else {
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(task);
```

Comment:
can refactor the code and extract out the System.out.println(task) and put it outside of the if else loop!